### PR TITLE
doc: Add MPI-4.0 partial compliance features

### PR DESCRIPTION
**Pull request by wckzhang**

### CODE REVIEW

File: docs/release-notes/mpi.rst [hunk 1]
```diff
@@ -31,6 +31,27 @@ For historical reference:
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the Open MPI |ompi_series| series, only partial MPI-4.0 functionality is
+supported. This section contains a list of features added for the release.
+
+* Support for MPI Sessions has been added.
+* Added partitioned communication using persistent sends and persistent receives.
+* Added persistent collectives to the ``MPI_`` namespace (they were previously
+  available via the ``MPIX_ prefix``).
+* Added ``MPI_Isendrecv()`` and its variants.
```
[the remark this answers is not in the document]
Comment (wckzhang) [reply docs/release-notes/mpi.rst:44]: Sure

File: docs/release-notes/mpi.rst [hunk 2]
```diff
@@ -31,6 +31,27 @@ For historical reference:
...
+* Added support for ``MPI_Comm_idup_with_info()``.
+* Added support for ``MPI_Info_get_string()``.
+* Added support for ``initial_error_handler`` and the ``ERRORS_ABORT`` infrastructure.
+* Added error handling for “unbound” errors to ``MPI_COMM_SELF``.
+* Made ``MPI_Comm_get_info()``, ``MPI_File_get_info()``, and
+  ``MPI_Win_get_info()`` compliant to the standard.
+* Droped unknown/ignored info keys on communicators, files, and windows.
+* Initial implementations of ``MPI_COMM_TYPE_HW_GUIDED`` and
```
[the remark this answers is not in the document]
Comment (wckzhang) [reply docs/release-notes/mpi.rst:52]: This was copied verbatim from the news - I can change it

**Comment by wckzhang** [comment]
Aye, I'll cross compare

**Comment by wckzhang** [comment]
also @jsquyres wzamazon is Wei, I'm wckzhang

**Comment by jsquyres** [comment]
> also @jsquyres wzamazon is Wei, I'm wckzhang

☹️ My bad.  Sorry about that!

**Comment by wckzhang** [comment]
@jsquyres updated

**Comment by jsquyres** [comment]
@wckzhang I made a few suggestions in a trailing commit on this PR.  If you accept them, squash them into your commit.

In making those suggestions, I found some other errors regarding the MPI standard major/minor version rendering, so I pre-pended some commits on this PR to fix that, too.  No need to squash those -- just wanted to make you aware of why they're here.

**Comment by wckzhang** [comment]
Squashed it, thanks Jeff

**Review by jsquyres — APPROVED** [review]

**Comment by wckzhang** [comment]
done - https://github.com/open-mpi/ompi/pull/11539

**Pull request merged**